### PR TITLE
[FX-4529] Add split button support to `Alert`

### PR DESCRIPTION
**Pull request by mkrl**
[FX-4529](https://toptal-core.atlassian.net/browse/FX-4529)

### Description

According to BASE, the action buttons in `Alert` should also support their split variants with dropdown menus. 
Each button can either be normal or split, with only two variants available (one of each at the time only).

### How to test

- [Temploy](https://picasso.toptal.net/fx-4529-extend-cta-button-in-the-alert-component-to-use-the-split-button-variant-as-well)
- See if the code and API looks good
- Look at the new examples for `Alert` on the temploy
- Tests should pass

### Screenshots

image

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4529]: https://toptal-core.atlassian.net/browse/FX-4529?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkrl** [comment]
@toptal-anvil ping reviewers

**Review by dmaklygin — APPROVED** [review]

File: .changeset/perfect-apricots-buy.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso': patch
```
[the remark this answers is not in the document]
Comment (sashuk) [reply .changeset/perfect-apricots-buy.md:2]: If we are adding some new functionality, it should be minor, right?

**Review by sashuk — APPROVED** [review]
Code looks good, tested on temploy!

File: .changeset/perfect-apricots-buy.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkrl) [reply .changeset/perfect-apricots-buy.md:2]: good point, might want to change that to minor!

**Pull request merged**